coder-sdk/workspace_providers.go

```diff
@@ -0,0 +1,80 @@
+package coder
+
+import (
+	"context"
+	"net/http"
+)
+
+// WorkspaceProvider defines an entity capable of deploying and acting as an ingress for Coder environments.
+type WorkspaceProvider struct {
+	ID                  string                  `json:"id"`
+	Name                string                  `json:"name"`
+	Status              WorkspaceProviderStatus `json:"status"`
+	Local               bool                    `json:"local"`
+	ClusterAddress      string                  `json:"cluster_address"`
+	DefaultNamespace    string                  `json:"default_namespace"`
+	StorageClass        string                  `json:"storage_class"`
+	ClusterDomainSuffix string                  `json:"cluster_domain_suffix"`
+	EnvproxyAccessURL   string                  `json:"envproxy_access_url" validate:"required"`
+	DevurlHost          string                  `json:"devurl_host"`
+	SSHEnabled          bool                    `json:"ssh_enabled"`
+	NamespaceWhitelist  []string                `json:"namespace_whitelist"`
+	OrgWhitelist        []string                `json:"org_whitelist"`
```

Comment:
These should all have `table` fields, with most hidden via `-`.

Reply:
They displayed nicely in my test, what does the table tags do exactly?

Reply:
If the table tag is not present, we default to the Go identifier name... which isn't exactly good behavior. Plus, the line length of the test shown above it far too large... we should be hiding `ID` and a few others.